### ISSUE TITLE
CLI to convert JSON to/from HCL

### ISSUE THREAD

**Issue by drnic** [hashicorp/hcl#13]
There seems to be support for JSON <--> HCL; a CLI that makes it easy to convert JSON into HCL would make it easy to integrate HCL into other tool chains?

**Comment by kvz** [comment]
Hey there, you can use the HCL printer (and `terraform fmt`) to do this now. @acconut and I took the liberty of wrapping a few of HashiCorp's libraries and release `json2hcl` (and hcl2json via `-reverse`) binaries for this. In the offchance this is still useful to somebody: https://github.com/kvz/json2hcl

**Comment by emind-johananl** [comment]
Thanks so much @kvz !

**Comment by jstoner** [comment]
@kvz this is great. We might use this to manage comparing parse trees for .tf files for different versions, and probably other things leveraging the json ecosystem.

**Comment by danielwpz** [comment]
@kvz Hey, I tried your tool to convert a nomad hcl job definition file to json and submitted it to nomad cluster, but nomad cannot accept the json output. Any idea what's the difference between terraform and nomad hcl?

**Comment by github4es** [comment]
Working with consul, our team has standardized on JSON.  The example https://www.consul.io/docs/connect/configuration.html . gives this snippet.   I've tried converting this into json but have had no luck. What is the json representation of this?

connect {
  enabled = true
}

**Comment by apparentlymart** [comment]
Hi all!

The exact mapping of HCL constructs to JSON depends on the expectations of the parsing program, due to the fact that e.g. some nested blocks can be specified multiple times (and thus map to a JSON array) while others can be specified only once (and so are better specified as a single object). Therefore for advice on how to write JSON input for a particular application, you're likely to get better answers from the team that works on that application, in cases where it's not already covered in the application docs.

They can hopefully also use your question/feedback to improve the application-specific documentation with more examples of the JSON variants of their respective configuration formats.
